### PR TITLE
Add block height to asset mint proof

### PR DESCRIPTION
**Pull request by ffranr**
Addresses this issue: https://github.com/lightninglabs/taproot-assets/issues/387

### CODE REVIEW

File: proof/records.go [hunk 1]
```diff
@@ -55,6 +56,12 @@ func BlockHeaderRecord(header *wire.BlockHeader) tlv.Record {
 	)
 }
 
+func BlockHeightRecord(height *uint32) tlv.Record {
```
Comment (guggero) [inline proof/records.go:59]: This commit needs to be earlier, otherwise the previous one doesn't compile.

Comment (guggero) [reply proof/records.go:59]: Or just squash this with the previous one, as there are some changes to the same line in here anyway.

File: taprpc/taprootassets.proto [hunk 2]
```diff
@@ -199,6 +199,9 @@ message AnchorInfo {
     anchor output.
     */
     bytes tapscript_sibling = 7;
+
+    // The height of the block which contains the anchor transaction.
+    uint32 block_height = 8;
```
Comment (guggero) [inline taprpc/taprootassets.proto:204]: We usually generate the protos in the same commit as we add/update fields.

File: proof/proof.go [hunk 3]
```diff
@@ -152,6 +151,7 @@ func (p *Proof) DecodeRecords() []tlv.Record {
 		MetaRevealRecord(&p.MetaReveal),
 		AdditionalInputsRecord(&p.AdditionalInputs),
 		ChallengeWitnessRecord(&p.ChallengeWitness),
+		BlockHeightRecord(&p.BlockHeight),
```
Comment (guggero) [inline proof/proof.go:154]: Should we add this field to one of the unit tests? I would assume that would've caught the "TLV stream must be canonical" error that caused you to move this field to the bottom.

**Review by guggero — COMMENTED** [review]
Looks good, just needs a small addition to a unit test.

File: taprpc/taprootassets.proto (hunk 2 above)
Comment (jharveyb) [reply taprpc/taprootassets.proto:204]: Also applies to the sqlc generate step as well IIRC.

File: proof/proof.go (hunk 3 above)
Comment (ffranr) [reply proof/proof.go:154]: In the latest commits, I've added the block height field into a proof encode/decode unit test.

**Comment by jharveyb** [comment]
Do we also want to return the confirmed block height when querying assets outside of the `genesis_info_view`?

One way would be to modify `QueryAssets` to also return the block height:

https://github.com/lightninglabs/taproot-assets/blob/22932e01e366334b9efb016a6c39c3edfe909ccb/tapdb/sqlc/queries/assets.sql#L372

And then update `dbAssetsToChainAssets` to populate the corresponding field in returned assets:

https://github.com/lightninglabs/taproot-assets/blob/22932e01e366334b9efb016a6c39c3edfe909ccb/tapdb/assets_store.go#L706

I don't think this actually blocks #367, but for testing it would be useful to have a query that accesses the `block_height` field added to `genesis_info_view`.

**Comment by ffranr** [comment]
> Do we also want to return the confirmed block height when querying assets outside of the genesis_info_view?

That sounds like a good idea @jharveyb . I'd like to merge this PR as is and make asset related changes in a different PR.
I think I'll make a different PR for the block height verification also.

**Review by guggero — APPROVED** [review]
LGTM :tada:

**Review by jharveyb — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**